Basic Swap V2

LeifuChen opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
Implementing a new v2 Swap UI into Kwenta: (basic) Synth Exchange.

## Related issue
#452

## Motivation and Context
Migrate more functions from V1 to V2.

## How Has This Been Tested?
Swap different tokens via UI.

## Screenshots (if appropriate):
![image](https://user-images.githubusercontent.com/4819006/162512301-875166ed-0589-44ff-9eb2-66eac49e11b9.png)

LeifuChen commented:
- [ ] 1) Navbar: Probably as soon as we merge Exchange, we should rename "Markets" to "Futures" ?
- [ ] 2a) Select Synths modal: Can we add a condition to hide the buttons for "Equities"/"Commodities" if there are 0 available?
- [ ] 2b) Select Synths modal: There are still a bunch of labels in  elderberry color, we probably want to change them to color: `${(props) => props.theme.colors.common.secondaryGray};` to be in line with the rest of the design
<p align="center">
  <img src="https://user-images.githubusercontent.com/4819006/162610390-a4959718-f555-4df0-baaa-bae5def795c1.png"  width="40%" />
</>

- [ ] 3) Just testing it I would have probably have liked it to have the "Select Synths" fields auto-filled on first visit with a default pair, e.g. sUSD-sETH, or maybe at least for "sUSD" to be set as first default maybe. But not sure that is desired, that's a question for product. Similar to Uniswap that shows default ETH and then second pair "Select token".
- [ ] 4) Same with color here, elderberry should be `secondaryGray`. Not sure Price Adjustment applies to L2, so maybe we can hide that info if network != Ethereum ?
<p align="center">
  <img src="https://user-images.githubusercontent.com/4819006/162610445-09de41bb-8fc5-4afe-b91a-15f706340032.png"  width="40%" />
</p>

- [ ] 5) More general question: Is this the default name for the synth by Synthetix? I think this should be `Synthetic Crude Oil (WTI)"` or something like this, or am I technically buying an Oil PERP here?
- [ ] 6) The arrow in the middle of the Synth Exchange modal, I noticed I wanted to click it to swap the two selected synths I had. Is this functionality desired by the design? 
For the rest, exchanging synths works like a charm, it looks super clean and just feels totally smooth to use. Great work!
- [ ] 7) When I select sUSD, and type an input, e.g. 100 ; and then second step "Select Synth", all the inputs reset to 0. I would want that the 100 sUSD in the first input won't reset, this kind of forces me to go up - down - up in my exchange flow
- [ ] 8) PR https://github.com/Kwenta/kwenta/pull/655 removes this check on dev (v1) because dHedge integration does not seem to work using this. We need to check this for v2-dev to make sure dHedge integration works without issues 

<p align="center">
  <img src="https://user-images.githubusercontent.com/4819006/162610473-cd18fc7f-be6a-41f0-8427-deba966f0a5a.png"  width="80%" />
</p>

LeifuChen commented:
@platschi I have made the changes based on your feedback (see the above tracklist). Please verify the functionalities. Thanks.

platschi commented:
Works absolutely great now, exchange works as expected. A few nitpicks design things:

- [ ] Make sure long Synth names have some spacing between lines, see:

<img width="373" alt="Screen Shot 2022-04-12 at 10 09 22" src="https://user-images.githubusercontent.com/548702/162969762-589f683e-5935-45de-88ca-404bc1537b51.png">

- [ ] Select Synths: Clicking a button, e.g. in below example "Commodities", works great, but as a user I have no way to know that I need to click one more time on the Commodities button to get back to the "All Synths" overview. I would suggest to implement a similar, subtle effect equal as for the "Long/Short" buttons on Futures Markets, e.g. using `transition: all 0.1s ease-in-out 0s` and `transform: scale(0.98)` or so. @asherism what do you think?

<img width="458" alt="Screen Shot 2022-04-12 at 10 10 27" src="https://user-images.githubusercontent.com/548702/162970263-35b50e1a-1e5a-4b45-a1e6-cb55dd7f35fb.png">

LeifuChen commented:
- [X] Adapt the long synth name in UI

<p align="center">
<img src="https://user-images.githubusercontent.com/4819006/163142496-c4cba710-3c26-421d-9439-18c5880f13b6.png" width="60%">
</p>

- [X] Add a new effect to show if the button is selected
<p align="center">
<img src="https://user-images.githubusercontent.com/4819006/163143103-2357ebe4-52a4-4f51-817c-ca2d43698706.png" width="60%">
</p>

LeifuChen commented:
@platschi The following changes have been made.
1. Updated the selected effect of the `categoryButton`
2. Removed "Synth Fees" in tooltip
3. Showed "Max Fee" on L1 and removed the "EDIT" button
4. Enabled the input of "Into".

Tburm commented on sections/shared/components/GasPriceSelect/GasPriceSelect.tsx:
```diff
@@ -36,22 +31,24 @@ const GasPriceSelect: FC<GasPriceSelectProps> = ({ gasPrices, transactionFee, ..
 	const [gasSpeed, setGasSpeed] = useRecoilState<keyof GasPrices>(gasSpeedState);
```
Can delete?

LeifuChen replied:
This is a duplicated comment :)

Tburm commented on sections/shared/components/GasPriceSelect/GasPriceSelect.tsx:
```diff
@@ -36,22 +31,24 @@ const GasPriceSelect: FC<GasPriceSelectProps> = ({ gasPrices, transactionFee, ..
 	const [gasSpeed, setGasSpeed] = useRecoilState<keyof GasPrices>(gasSpeedState);
 	const [customGasPrice, setCustomGasPrice] = useRecoilState(customGasPriceState);
```
Can delete?
```suggestion
```

LeifuChen replied:
I removed the set method, which is not required.

Tburm commented on sections/shared/components/GasPriceSelect/GasPriceSelect.tsx:
```diff
@@ -36,22 +31,24 @@ const GasPriceSelect: FC<GasPriceSelectProps> = ({ gasPrices, transactionFee, ..
 	const [gasSpeed, setGasSpeed] = useRecoilState<keyof GasPrices>(gasSpeedState);
```
Can delete?
```suggestion
```

LeifuChen replied:
I removed the set method, which is not required.

Tburm reviewed (changes requested):
Balance is a little crowded with a large balance 🐳  :
<img width="187" alt="image" src="https://user-images.githubusercontent.com/10401554/163452723-35516033-efe0-49aa-8ebc-65316fa4d857.png">

When cancelling a transaction, we probably want more padding on this card/button:
<img width="308" alt="image" src="https://user-images.githubusercontent.com/10401554/163453816-81353f25-f0e0-47f8-8ed1-9d1650e9ac1e.png">

I had a test transaction fail with "out of gas" on Optimism Kovan:
* [tx](https://kovan-optimistic.etherscan.io/tx/0x430a581e87ae0aae4376ddea2a8975ec5de44f372c670fa6e344d3aae2dc2164)
* [tenderly](https://dashboard.tenderly.co/tx/optimistic-kovan/0x430a581e87ae0aae4376ddea2a8975ec5de44f372c670fa6e344d3aae2dc2164)

After a transaction succeeds, it's really nice that balances are updated. It would also be good to clear the inputs to make it clear the transaction went through.

For assets with high value (sBTC) we should add more decimals:
<img width="387" alt="image" src="https://user-images.githubusercontent.com/10401554/163455576-6b6c3467-5dd2-4c82-8063-4ca113f23a33.png">

Tburm commented on sections/exchange/TradeCard/CurrencyCard/CurrencyCard.tsx:
```diff
@@ -246,27 +322,44 @@ const Slippage = styled.div`
 		css`
 			pointer-events: none;
 		`}
+	width: 136px;
```
```suggestion
	width: 100%;
```

LeifuChen replied:
Increased the width of  `StyledBaseModal` to 300px.

LeifuChen commented:
@Tburm  @platschi  The following issues have been fixed.
1. Update the precision of select currency modal
2. Increase the width of TxConfirmationModal
3. Fix the wallet balance display for whales
4. Remove the unnecessary set method
5. Reset the input fields after the confirmation of the trades
6. "Into" input fields only update when a synth selected
